org.artisan_scope.artisan.yml

```diff
@@ -0,0 +1,107 @@
+id: org.artisan_scope.artisan
+runtime: org.kde.Platform
+sdk: org.kde.Sdk
+runtime-version: '6.7'
+
+base: com.riverbankcomputing.PyQt.BaseApp
+base-version: '6.7'
+cleanup-commands:
+  # uninstall build dependencies
+  - python3 -m pip uninstall -y meson-python cppy pybind11
+  # deduplicate (matplotlib) fonts already in runtime
+  - for f in /usr/share/fonts/dejavu/*.ttf; do find /app -name `basename $f` -execdir ln -sf "$f" \; ; done
+  # remove Python test suites
+  - find $FLATPAK_DEST/lib/python* -type d -name tests | xargs rm -Rf
+  # Yoctupuce includes compiled libraries *sigh*, only keep the ones we may need
+  - |
+    python3 <<EOF
+    from os import listdir, remove
+    from os.path import basename, dirname, join
+    from yoctopuce.yocto_api import YAPI
+    YAPI.yloadYapiCDLL()
+    dir = dirname(YAPI._yApiCLibFile)
+    keep = [basename(f) for f in [YAPI._yApiCLibFile]] # might include YAPI._yApiCLibFileFallback
+    [remove(join(dir, f)) for f in listdir(dir) if f not in keep]
+    EOF
+  # only keep babel and QtWebEngine locales that the app is translated into
+  - |
+    LANGRE='\('`ls $FLATPAK_DEST/translations/*.qm | sed 's/^.*artisan_\([a-zA-Z]\+\)\(_[a-zA-Z]\+\)\?\.qm$/\1/' | grep -v '^qtbase_' | sort | uniq | tr '\n' '|' | sed 's/|/\\\\|/g'`'en\)'
+    (cd $FLATPAK_DEST/translations/qtwebengine_locales/ && ls *.pak | grep -v "^$LANGRE\([-_][a-zA-Z]\+\)\?\.pak$" | xargs rm -f )
+    (cd $FLATPAK_DEST/lib/python3*/site-packages/babel/locale-data/ && ls *.dat | grep -v "^$LANGRE\([-_][-_a-zA-Z]\+\)\?\.dat$" | xargs rm -f )
+  # cleanup QtWebEngine dictionaries, which we don't need and use quite some space
+  - find $FLATPAK_DEST/share/locale -name \*.bdic | xargs rm -f
+  - rm -Rf $FLATPAK_DEST/qtwebengine_dictionaries
+  # we need no QtWebEngine developer tools, nor automation driver
+  - rm -f $FLATPAK_DEST/resources/qtwebengine_devtools_resources.pak
+  - rm -Rf $FLATPAK_DEST/libexec/webenginedriver
+  # cleanup base
+  - /app/cleanup-BaseApp.sh
+
+command: artisan
+separate-locales: false
+finish-args:
+  # Qt
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  #- --device=dri # we already have --device=all
+  # USB
+  - --device=all
+  # allow network access for network devices and artisan.plus
+  - --share=network
+  # storing and retrieving artisan.plus credentials
+  - --talk-name=org.freedesktop.secrets
+  # filesystem access needed for autosave
+  - --filesystem=xdg-documents
```

Comment:
I tried making it work without xdg-documents access, but couldn't get autosave to work. Some things depend on the version of the xdg desktop portal in use, and in theory even opening a directory should work, but I never got it working. To make this work for various users, I chose the xdg-documents approach, allowing autosave to work in the user's Documents directory, but not expose all files. I've seen this in other flatpaks as well, and hope this line can be removed at some point.